### PR TITLE
a5 onc 2015

### PR DESCRIPTION
**Pull request by bradymiller**
a5 item for onc 2015

continued from https://github.com/openemr/openemr/pull/3266

completed:
Added a new datatype that allows a comment to be included with list
Added a Unknown birth sex
Change the ethnicity and race datatype to use multselect list rather than single select list

### CODE REVIEW

**Comment by sunsetsystems** [comment]
Reminds me of an idea for a SOC project: Move Demographics and History data items to vertical tables, as is done with encounter LBFs.

File: interface/super/edit_layout.php [hunk 1]
```diff
@@ -84,6 +84,7 @@ function collectLayoutNames($condition, $mapping = '')
   "43" => xl("List box w/search"),
   "44" => xl("Multi-Select Facilties"),
   "45" => xl("Multi-Select Provider"),
+  "46" => xl("List box w/comment"),
```
Comment (sjpadgett) [inline interface/super/edit_layout.php:87]: I'm guessing nothing special needs to be handled in LBF/new.php

**Comment by sunsetsystems** [comment]
Sorry to notice this so late... did you know that conditions in layout items can be used to hide or show a field depending on the value of another field? That might be a way to do what you want instead of creating a new data type.

**Comment by sjpadgett** [comment]
which is very useful for questionnaires. now we just need to sum items:)

**Comment by sunsetsystems** [comment]
> which is very useful for questionnaires. now we just need to sum items:)

LBF plug-ins can be used for more esoteric things like that. :)

**Comment by bradymiller** [comment]
@sunsetsystems , thanks for the pointer. i'll check out the condition stuff to see if can avoid the new datatype.

**Comment by bradymiller** [comment]
This PR will now officially pass ONC2015 item a5 :)
Here's what this PR does:
1. Added a new layout datatype that allows a comment to be included with list
2. Used the new datatype to support Gender Identity and Sexual Orientation selectors in demographics
3. Added a Unknown birth sex option
4. Changed the ethnicity and race datatype to use multselect list rather than single select list

Still some more road testing to do.

@sunsetsystems , did also try the Gender Identity and Sexual Orientation comment support via layout conditionals, which I could get to work. In this case, though, I like the idea of having the list determine which element(s) get a comment rather than relying on the layout entry that uses the list. This also ensure that it won't get broken by users that modify layouts since it is a core piece of ONC 2015. And now the user has 2 potential mechanisms to support this kind of feature. In the clinical realm (ie. LBF stuff), it is really common to have selectors to choose from where some of the options require a comment, so having this new datatype will be another good option in those scenarios. Hopefully that convinces you to embrace the new datatype :)

**Comment by sunsetsystems** [comment]
Brady, consider it embraced. Just wanted to make sure you know about conditionals. Thanks for the improvements!

**Comment by bradymiller** [comment]
testing well and bringing in

**Comment by stephenwaite** [comment]
excited_elephant

**Comment by sjpadgett** [comment]
Puppies are cute, kittens are cute... baby elephants, not so much.:)
Check out bird in background saying to itself: "yeah, i'm remember when I was young once!"

**Comment by stephenwaite** [comment]
had a lot of questions about this functionality for a few years, thanks again for the improvements @bradymiller !

**Pull request merged**